### PR TITLE
[client/py] - added type coersion for primitive non-string values in …

### PR DESCRIPTION
**Pull request by emilbon99**
…range KV

# Feature Pull Request Template

## Key Information

- **Linear Issue**: [SY-1452](https://linear.app/synnax/issue/SY-1452/add-auto-conversion-to-string-for-metadata-in-python)

## Description

Adds type coercion in the Python client for non-string values on the range key-value store. 

## Basic Readiness

- [x] I have performed a self-review of my code.
- [x] I have added relevant tests to cover the changes to CI.
- [ ] I have needed QA steps to the [release
      candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template that cover these changes.
- [ ] I have updated in-code documentation to reflect the changes.
- [ ] I have updated user-facing documentation to reflect the changes.

## Backwards Compatibility

The following makes sure that this feature does not break backwards compatability.

### Data Structures

- [ ] Server - I have ensured that previous versions of stored data structures are
      properly migrated to new formats.
- [ ] Console - I have ensured that previous versions of stored data structures are
      properly migrated to new formats.

### API Changes

- [ ] Server - The server API is backwards-compatible
- The following client APIs are backwards-compatible:
  - [ ] C++
  - [ ] TypeScript
  - [ ] Python

### Breaking Changes

If anything in this section is not true, please list all breaking changes.

### CODE REVIEW

File: client/py/synnax/ranger/kv.py [hunk 1]
```diff
@@ -12,14 +12,27 @@
...
+from synnax.util.primitive import Primitive, is_primitive
 
 
 class KVPair(Payload):
     range: uuid.UUID
     key: str
     value: str
 
+    def __init__(self, **kwargs):
+        value = kwargs.get("value")
```
Comment (pjdotson) [inline client/py/synnax/ranger/kv.py:26]: Isn't the better solution here to check if an object has a `__str__()` method instead? No need to restrict ourselves to primitives.

Comment (pjdotson) [reply client/py/synnax/ranger/kv.py:26]: Or call the `str` function and catch the exception and throw our own

Comment (emilbon99) [reply client/py/synnax/ranger/kv.py:26]: The problem is that all python objects have a __str__ method. But classes have a representation like <__main__.MyClass object at 0x000001AB2C8C7A90>, which I don't think we should allow

Comment (pjdotson) [reply client/py/synnax/ranger/kv.py:26]: You could check if they use the default object, ChatGPT gives something like this:

```python
def safe_str(obj):
    # Check if the object's __str__ method is the default one from `object`
    if type(obj).__str__ == object.__str__:
        raise TypeError(f"{type(obj).__name__} has not overridden the default __str__ method.")
    return str(obj)

# Example usage:
class MyClass1:
    pass

class MyClass2:
    def __str__(self):
        return "This is a custom string representation."

obj1 = MyClass1()
obj2 = MyClass2()

try:
    print(safe_str(obj1))
except TypeError as e:
    print(f"Error: {e}")

try:
    print(safe_str(obj2))
except TypeError as e:
    print(f"Error: {e}")
```

Comment (emilbon99) [reply client/py/synnax/ranger/kv.py:26]: Yes we could do this, but what about custom stringers from imported libraries? Should we not error or warn on those?

Don't have a tremendously strong opinion here

Comment (emilbon99) [reply client/py/synnax/ranger/kv.py:26]: For example, the `WebSocket` constructor has a custom stringer that overrides the default one - is that acceptable? I guess so

Comment (pjdotson) [reply client/py/synnax/ranger/kv.py:26]: Yeah I think it is, that case seems relatively rarer and this is generally the better solution. This is kind of the "be liberal in what you accept, conservative in what you output" principle.

Comment (emilbon99) [reply client/py/synnax/ranger/kv.py:26]: Ok, will change

**Review by pjdotson — APPROVED** [review]

**Pull request merged**